mesonbuild/modules/fs.py

```diff
@@ -312,6 +316,20 @@ def copyfile(self, state: ModuleState, args: T.Tuple[FileOrString, T.Optional[st
 
         return ModuleReturnValue(ct, [ct])
 
+    @FeatureNew('fs.relative_to', '1.3.0')
+    @typed_pos_args('fs.relative_to', (str, File), (str, File))
+    @noKwargs
+    def relative_to(self, state: ModuleState, args: T.Tuple[FileOrString, FileOrString], kwargs: TYPE_kwargs) -> str:
+        t = str(args[0])
+        f = str(args[1])
```

Comment:
I think str(File) returns subdir/file, and you'll add subdir again. can you please add a unit test that mix File and str to verify that?

Reply:
Is there anyway to get the build root from ModuleState? File::absolute_path() requires passing both the source root and the build root.